packages/data-flow/src/registries/event/cachedEventRegistry.ts

```diff
@@ -0,0 +1,70 @@
+import { NewProcessedEvent, ProcessedEvent } from "@grants-stack-indexer/repository";
+import { ChainId, ILogger, stringify } from "@grants-stack-indexer/shared";
+
+import { IEventsRegistry } from "../../internal.js";
+
+/**
+ * Proxy class to cache the events in memory or fallback to another event registry
+ */
+export class InMemoryCachedEventRegistry implements IEventsRegistry {
+    private cache: Map<ChainId, ProcessedEvent> = new Map();
+
+    private constructor(
+        private readonly logger: ILogger,
+        private readonly eventRegistry: IEventsRegistry,
+        cache: Map<ChainId, ProcessedEvent>,
```

Comment:
Extremely minor, but maybe renaming it to `initialEntries` or `initialCache` and making it optional would make this a little bit more explanatory?

```suggestion
        initialCache?: Map<ChainId, ProcessedEvent>,
```

Nothing serious though, feel free to dismiss this comment if you prefer it the way it is.

Reply:
the goal of this class this is a proxied cache so it always is initialized with a pre-filled cache that fallbacks to the main registry implementation and given that constructor is private i think it's the same to have it optional or not